client/components/application-summary.js

```diff
@@ -205,6 +251,109 @@ const ApplicationSummary = () => {
+   * @returns {boolean} - True if any field differs, false otherwise.
+   */
+  const compareFields = (data) => {
+    let changed = false;
+
+    Object.entries(data).forEach(([key, value]) => {
+      // Handle nested objects (e.g., subsections)
+      if (value && typeof value === "object" && !Array.isArray(value)) {
+        if (compareFields(value)) {
+          changed = true;
+        }
+      } else {
+        // Compare leaf fields
+        const savedValue = savedValues[key];
+        const currentValue = currentValues[key];
+        const initialValue = initialValues[key];
+
+        const sanitisedSaved    =  cleanValue(savedValue);
+        const sanitisedCurrent  =  cleanValue(currentValue);
+        const sanitisedInitial  =  cleanValue(initialValue);
+
+        if (
+          sanitisedSaved !== sanitisedCurrent ||
+          sanitisedSaved !== sanitisedInitial
+        ) {
+          changed = true;
+        }
+      }
+    });
+
+    return changed;
+  };
+
+  return compareFields(sectionData);
+};
+
+
+
+
+/**
+ * Determines if any fields in a given set have changed by comparing their current and initial values.
+ * 
+ * This function is designed to handle various field types, including:
+ * 
+ * - **Text & textarea fields:** Compared as normalised and sanitised strings.
+ * - **Checkboxes & multi-selects:** Compared as sorted arrays to ensure order does not affect the result.
+ * - **Radio buttons & boolean fields:** Directly compared as boolean values.
+ * 
+ * The function iterates through all specified fields and returns `true` if any have changed.
+ *
+ * @param {Array} fields - The list of field names to check for changes.
+ * @param {Object} currentValues - The current values of the fields, typically from user input or state.
+ * @param {Object} initialValues - The initial values of the fields, typically from the database or original state.
+ * @returns {boolean} - Returns `true` if at least one field has changed, otherwise `false`.
+ */
+const hasChangedFields = (fields, currentValues, initialValues) => {
+  return fields.some(field => {
+    // Normalise and sanitise current and initial values
+    const currentValue = cleanValue(currentValues[field]);
+    const initialValue = cleanValue(initialValues[field]);
+
+    // Special handling for array-based fields (e.g., checkboxes, multi-selects)
+    if (Array.isArray(currentValue) || Array.isArray(initialValue)) {
+      const currentArray = Array.isArray(currentValue) ? currentValue : [];
+      const initialArray = Array.isArray(initialValue) ? initialValue : [];
+      const changed = JSON.stringify(currentArray.sort()) !== JSON.stringify(initialArray.sort());
+
+      return changed;
+    }
+
+    // Special handling for boolean fields (e.g., checkboxes, radios)
+    if (typeof currentValue === "boolean" || typeof initialValue === "boolean") {
+      const changed = currentValue !== initialValue;
+
+      return changed;
+    }
```

Comment:
After normalisation current value and initial value are always strings so this is redundant and misleading.
```suggestion
```